app/services/people/import_from_nomis.rb

```diff
@@ -12,43 +12,53 @@ class ImportFromNomis
       prison_number: :prison_number,
     }.freeze
 
-    def initialize(prison_number)
-      @prison_number = prison_number
+    def initialize(prison_numbers)
+      @prison_numbers = prison_numbers
     end
 
     def call
-      person = Person.find_or_initialize_by(prison_number: @prison_number)
-      person.assign_attributes(person_attributes)
-      person.save!
+      @prison_numbers.each do |prison_number|
+        person = Person.find_or_initialize_by(prison_number: prison_number)
+
+        nomis_person = people_from_nomis.find do |nomis_person|
+          nomis_person[:prison_number] == prison_number
+        end
+
+        next unless nomis_person
```

Comment:
I think we can make this clearer, we could filter @prison_number before looping.